qCC/ccRegistrationTools.cpp

```diff
@@ -194,7 +199,7 @@ bool ccRegistrationTools::ICP(	ccHObject* data,
 			ParallelSort(distances.begin(), distances.end());
 			
 			//now look for the max value at 'finalOverlapRatio+margin' percent
-			maxSearchDist = distances[static_cast<unsigned>(std::max(1.0,count*(finalOverlapRatio+s_overlapMarginRatio)))-1];
+			maxSearchDist = distances[static_cast<unsigned>(std::max(1.0,count*(static_cast<double>(finalOverlapRatio)+s_overlapMarginRatio)))-1];
```

Comment:
Isn't finalOverlapRatio already a double value?

Reply:
yes I was thrown off by the squigles going away for the static code analyzer it takes a moment after saving before showing any warnings, but the (Warning C26451: Arithmetic overflow) was actually for the static_cast<unsigned>
removing my first attempt, and changing to  

``` C++
maxSearchDist = distances[static_cast<size_t>(std::max(1.0,count*(finalOverlapRatio+s_overlapMarginRatio)))-1];
```
cleared it up. 
unsigned = 4 bytes, size_t = 8 bytes

Is there any reason to not use size_t in this case?

Reply:
just started using VS2019 and I'm getting a ton of static analysis warnings that weren't present in VS2017 especially everything QT related 
![image](https://user-images.githubusercontent.com/7495752/68508761-0f7ab680-0224-11ea-8691-0df873bf8740.png)

there is an open issue for [VS2019](https://developercommunity.visualstudio.com/content/problem/520393/vs-2019-detects-many-false-warnings-while-vs2017-w.html) 

but outside of the QT noise it does pickup some seemingly legit risks
![image](https://user-images.githubusercontent.com/7495752/68508942-9d56a180-0224-11ea-8602-61903f37b767.png)

Reply:
Originally CC was 32bits only (hence the unsigned everywhere) and we also wanted to keep point indexes on 32bits so as to minimize the memory consumption (so unsigned became the default integer type I would say). 

Here I assume that count is an unsigned? At least the result of the calculus can't be greater than 'count' but the compiler/Visual can't be sure and raise the warning by default I guess.

Reply:
The next example is indeed interesting. Here it's more about the loss of numerical accuracy 'for nothing'. The float values resulting from the subtractions can definitely be casted to double.